python/ray/data/_internal/plan.py

```diff
@@ -381,7 +383,14 @@ def schema(
         """
         from ray.data._internal.stage_impl import RandomizeBlocksStage
 
+        if self._schema is not None:
+            return self._schema
+
         if self._stages_after_snapshot:
+            # TODO(swang): There are several other stage types that could
```

Comment:
Yeah, was wondering if this should be a feature of the LogicalPlan API (cc @raulchen ).